Sources/SwiftParser/TokenSpecSet.swift

```diff
@@ -339,6 +428,80 @@ enum PureDeclarationKeyword: TokenSpecSet {
   }
 }
 
+enum MisspelledPureDeclarationKeyword: MisspelledTokenSpecSet {
+  case `associatedtype`
+  case `class`
+  case `deinit`
+  case `func`
+  case `init`
+  case `precedencegroup`
+  case `protocol`
+  case `typealias`
+
+  var correctSpecSet: PureDeclarationKeyword {
+    switch self {
+    case .associatedtype: return .associatedtype
+    case .class: return .class
+    case .deinit: return .deinit
+    case .func: return .func
+    case .`init`: return .`init`
+    case .precedencegroup: return .precedencegroup
+    case .protocol: return .protocol
+    case .typealias: return .typealias
+    }
+  }
+
+  init?(lexeme: Lexer.Lexeme, experimentalFeatures: Parser.ExperimentalFeatures) {
+    let text = lexeme.tokenText
+    switch text.count {
+    case 3:
+      switch text {
+      case "def": self = .func
+      case "fun": self = .func
+      default: return nil
+      }
+    case 6:
+      switch text {
+      case "deInit": self = .deinit
+      case "object": self = .class
+      default: return nil
+      }
+    case 8:
+      switch text {
+      case "function": self = .func
+      default: return nil
+      }
```

Comment:
I think we could use code generation in the future for creating matching statements like these.